azurerm/internal/services/cdn/cdn_endpoint_resource.go

```diff
@@ -303,7 +304,12 @@ func resourceArmCdnEndpointCreate(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error retrieving CDN Endpoint %q (Profile %q / Resource Group %q): %+v", name, profileName, resourceGroup, err)
 	}
 
-	d.SetId(*read.ID)
+	id, err := parse.CdnEndpointID(*read.ID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(id.ID(subscriptionId))
```

Comment:
we'll also need a state migration to update existing resources - see [the frontdoor migrations](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/frontdoor_custom_https_configuration_resource.go#L69-L76) for an example